### PR TITLE
confu_json:  parsing  json to user defined type and vice versa

### PR DESCRIPTION
**Pull request by werto87**
Specify library name and version:  **confu_json/0.0.5**

# confu_json
## How to:
  Define a boost fusion struct "https://www.boost.org/doc/libs/1_76_0/libs/fusion/doc/html/fusion/adapted/define_struct.html".
  
  Use to_json(object from your type) to create a json with the data from your type.
  
  Use to_object< your type >(json) to create an object from a json.
  
  
## Example:
  ```c++
  #include "confu_json/to_json.hxx"
  #include "confu_json/to_object.hxx"
  #include <boost/fusion/adapted/struct/adapt_struct.hpp>
  #include <boost/json/src.hpp>  // this file should be included only in one translation unit

  BOOST_FUSION_DEFINE_STRUCT((shared_class), Nested, (long, answer)) // used to define a struct

  int main() {
    using namespace confu_json;
    auto nested = shared_class::Nested{};
    nested.answer = 42;
    std::cout << to_json(nested) << std::endl; // converts the struct into json and prints it
    auto nestedTest = to_object<shared_class::Nested>(to_json(nested)); // converts the struct into json and back into an object
    assert(nested.answer == nestedTest.answer);
  }  
  ```
  Exampls for defining types: "https://github.com/werto87/confu_json/blob/main/test/constant.hxx"
  
  Exampls for usage: "https://github.com/werto87/confu_json/blob/main/test/combineToJsonAndToObject.cxx"

## This project builds on:
boost::json -> json handling 

boost::mpl and boost::fusion -> template metaprogramming

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/confu_json/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "10",
+            "clang": "10",
+        }
+
+    def configure(self):
```
Comment (prince-chrismc) [inline recipes/confu_json/all/conanfile.py:34]:
```suggestion
    def validate(self):
```

File: recipes/confu_json/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        self.options["boost"].header_only = True
+
+    def requirements(self):
+        self.requires("boost/1.76.0")
+        self.requires("magic_enum/0.7.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
```
Comment (prince-chrismc) [inline recipes/confu_json/all/conanfile.py:63]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version],
                  destination=self._source_subfolder, strip_root=True)
```

File: recipes/confu_json/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+from conans import ConanFile, tools
+from conans.tools import check_min_cppstd
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+from conans.tools import Version
+
+
+class ConfuJson(ConanFile):
```
Comment (prince-chrismc) [inline recipes/confu_json/all/conanfile.py:8]:
```suggestion
from conans.tools import Version

required_conan_version = ">=1.33.0"


class ConfuJson(ConanFile):
```

File: recipes/confu_json/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                                 self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(
+                    "{} requires C++{} support. "
+                    "The current compiler {} {} does not support it.".format(
+                        self.name, self._minimum_cpp_standard,
+                        self.settings.compiler,
+                        self.settings.compiler.version))
+        self.options["boost"].header_only = True
```
Comment (prince-chrismc) [inline recipes/confu_json/all/conanfile.py:54]: Let's not surprises the consumer 

```suggestion

    def configure(self):
        if not self.options["boost"].header_only:
            raise ConanInvalidConfiguration("{} requires a header-only Boost. ".format(self.name))
```

File: recipes/confu_json/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.5)
```
Comment (prince-chrismc) [inline recipes/confu_json/all/test_package/CMakeLists.txt:1]: `cxx_Std_20` was added later.... this should be a hook

```suggestion
cmake_minimum_required(VERSION 3.12)
```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**